Source/WTF/wtf/Int128.h

```diff
@@ -1264,10 +1271,72 @@ using UInt128 = UInt128Impl;
 using Int128 = Int128Impl;
 #endif
 
+template<> struct DefaultHash<UInt128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const UInt128& a, const UInt128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+template<> struct DefaultHash<Int128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const Int128& a, const Int128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+#if !HAVE(INT128_T)
+template<> struct HashTraits<UInt128> : public SimpleClassHashTraits<UInt128> { };
+template<> struct HashTraits<Int128> : public SimpleClassHashTraits<Int128> { };
+#endif
```

Comment:
We should make them based on `GenericHashTraits` since we should not treat UInt128 / Int128 as a class.